perftestnotebook.py

```diff
@@ -144,66 +158,94 @@ def process(self):
 
         self.fmt_data = fmt_data
 
+        """
+        Write formatted data output to filepath
+        """
+        print(json.dumps(self.fmt_data, indent=4, sort_keys=True))
+
+        prefix = "output" if "prefix" not in self.config else self.config["prefix"]
+        output_data_filepath = "%s_fmt_data.json" % prefix
+
+        if "output" in self.config:
+            output_data_filepath = self.config["output"]
+
+        print("Writing results to %s" % output_data_filepath)
+
+        with open(output_data_filepath, "w") as f:
+            json.dump(self.fmt_data, f, indent=4, sort_keys=True)
+
+        """
+        Analyze data
+        """
+        analyzer_templates = ""
         if "analysis" in self.config:
-            # Analyze the data
-            all_results = {}
-            self.analyzer.data = fmt_data
+            
+            print(self.config["analysis"])
+            # without readLine
             for func in self.config["analysis"]:
-                all_results[func] = getattr(self.analyzer, func)()
-            return all_results
+                analyzer_templates = analyzer_templates + self.analyzer.get_analysis_template(func)
+            
+
+        """
+        Post to iodide server
+        """
+        if not no_iodide:
+           self.post_to_iodide(output_data_filepath,analyzer_templates)
 
         return self.fmt_data
 
+    def post_to_iodide(self, output_data_filepath,analyzer_templates):
+        """# Upload template file to Iodide"""
+        print(output_data_filepath)
+        template_header_path = "testing/resources/template/header"
+        template_content = ""
 
-def main():
-    args = parse_args()
+        with open(template_header_path, "r") as f:
+            template_content = template_content + f.read()
```

Comment:
You can do `template_content = f.read()` here, and below (it can be moved outside the `with` and changed to `template_content += notebook_sections`